pkg/middleware/middleware.go

```diff
@@ -33,6 +33,7 @@ func AddDefaultResponseHeaders(cfg *setting.Cfg) web.Handler {
 	t := web.NewTree()
 	t.Add("/api/datasources/uid/:uid/resources/*", nil)
 	t.Add("/api/datasources/:id/resources/*", nil)
+	t.Add("/api/plugins/:id/resources/*", nil)
```

Comment:
The change makes sense to me, to follow the same approach than datasources. I am not able to make it work with the example you point out. With this change `resourceURLMatch` is `true` but `resourceCachable` is still resolved to `false`. How do you verify this work with a real example?

Also, can you add a test for this in `middleware_test.go`?

Reply:
> resourceCachable is still resolved to false.

Thats fine. `resourceCachable` flips to true, once the resource handler `X-Grafana-Cache: true` and `Cache-Control: public`

Reply:
@andresmgot I have added tests and they are green.